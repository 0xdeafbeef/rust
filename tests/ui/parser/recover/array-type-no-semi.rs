// when the next token is not a semicolon,
// we should suggest to use semicolon if recovery is allowed
// See issue #143828

fn main() {
    let x = 5;
    let b: [i32, 5];
    //~^ ERROR expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
    //~| ERROR expected value, found builtin type `i32` [E0423]
    let a: [i32, ];
    //~^ ERROR expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
    //~| ERROR expected value, found builtin type `i32` [E0423]
    let c: [i32, x];
    //~^ ERROR expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
    //~| ERROR expected value, found builtin type `i32` [E0423]
    let e: [i32 5];
    //~^ ERROR expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `5`
}
