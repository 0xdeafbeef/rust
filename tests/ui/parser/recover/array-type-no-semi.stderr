error: expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
  --> $DIR/array-type-no-semi.rs:7:16
   |
LL |     let b: [i32, 5];
   |          -     ^ expected one of 7 possible tokens
   |          |
   |          while parsing the type for `b`
   |          help: use `=` if you meant to assign

error: expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
  --> $DIR/array-type-no-semi.rs:10:16
   |
LL |     let a: [i32, ];
   |          -     ^ expected one of 7 possible tokens
   |          |
   |          while parsing the type for `a`
   |          help: use `=` if you meant to assign

error: expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `,`
  --> $DIR/array-type-no-semi.rs:13:16
   |
LL |     let c: [i32, x];
   |          -     ^ expected one of 7 possible tokens
   |          |
   |          while parsing the type for `c`
   |          help: use `=` if you meant to assign

error: expected one of `!`, `(`, `+`, `::`, `;`, `<`, or `]`, found `5`
  --> $DIR/array-type-no-semi.rs:16:17
   |
LL |     let e: [i32 5];
   |          -      ^ expected one of 7 possible tokens
   |          |
   |          while parsing the type for `e`

error[E0423]: expected value, found builtin type `i32`
  --> $DIR/array-type-no-semi.rs:7:13
   |
LL |     let b: [i32, 5];
   |             ^^^ not a value

error[E0423]: expected value, found builtin type `i32`
  --> $DIR/array-type-no-semi.rs:10:13
   |
LL |     let a: [i32, ];
   |             ^^^ not a value

error[E0423]: expected value, found builtin type `i32`
  --> $DIR/array-type-no-semi.rs:13:13
   |
LL |     let c: [i32, x];
   |             ^^^ not a value

error: aborting due to 7 previous errors

For more information about this error, try `rustc --explain E0423`.
